HW_9/quadratic_equation_solver.py

```diff
@@ -0,0 +1,139 @@
+'''
+Напишите следующие функции:
+Нахождение корней квадратного уравнения
+Генерация csv файла с тремя случайными числами в каждой строке. 100-1000 строк.
+Декоратор, запускающий функцию нахождения корней квадратного
+уравнения с каждой тройкой чисел из csv файла.
+Декоратор, сохраняющий переданные параметры и результаты работы
+функции в json файл
+'''
+
+import csv
+import json
+import random
+from functools import wraps
+import math
+
+# Функция нахождения корней квадратного уравнения
+
+
+def solve_quadratic_equation(a, b, c):
+    """
+    Функция для нахождения корней квадратного уравнения.
+
+    Args:
+        a (float): Коэффициент a.
+        b (float): Коэффициент b.
+        c (float): Коэффициент c.
+
+    Returns:
+        tuple: Корни квадратного уравнения.
+    """
+    discriminant = b**2 - 4*a*c
+    if discriminant < 0:
+        return None
+    elif discriminant == 0:
+        x = -b / (2*a)
+        return x, x
+    else:
+        x1 = (-b + math.sqrt(discriminant)) / (2*a)
+        x2 = (-b - math.sqrt(discriminant)) / (2*a)
+        return x1, x2
+
+# Функция генерации CSV-файла с тремя случайными числами в каждой строке
+
+
+def generate_csv_file(filename):
+    """
+    Функция для генерации CSV-файла с тремя случайными числами в каждой строке.
+
+    Args:
+        filename (str): Имя файла для сохранения.
+    """
+    rows = random.randint(
+        100, 1000)  # Случайное количество строк от 100 до 1000
+    with open(filename, 'w', newline='') as file:
+        writer = csv.writer(file)
+        for _ in range(rows):
+            row = [random.random() for _ in range(3)]
+            writer.writerow(row)
+
+# Декоратор для запуска функции нахождения корней квадратного уравнения с каждой тройкой чисел из CSV-файла
+
+
+def solve_quadratic_equation_with_csv(func):
+    """
+    Декоратор для запуска функции нахождения корней квадратного уравнения с каждой тройкой чисел из CSV-файла.
+
+    Args:
+        func (function): Декорируемая функция.
+
+    Returns:
+        function: Обернутая функция.
+    """
+    @wraps(func)
+    def wrapper(filename):
+        with open(filename, 'r') as file:
+            reader = csv.reader(file)
+            for row in reader:
+                numbers = [float(num) for num in row]
+                func(*numbers)
```

Comment:
Это можно сделать сразу
```suggestion
            for row in reader:
                func(*(map(int, row)))
```